src/catalog/src/durable/persist.rs

```diff
@@ -1794,13 +1827,28 @@ impl UnopenedPersistCatalogState {
                 .map(|((k, v), _, _)| (T::update(k, v), -1))
                 .collect();
             updates.push((T::update(key, value), 1));
-            updates.extend(fence_updates.clone());
-            match self.compare_and_append(updates).await {
-                Ok(_) => break prev_value,
-                Err(CompareAndAppendError::Fence(e)) => return Err(e.into()),
-                Err(e @ CompareAndAppendError::UpperMismatch { .. }) => {
-                    warn!("catalog write failed due to upper mismatch, retrying: {e:?}");
-                    continue;
+
+            // We must fence out all other catalogs, if we haven't already, since we are writing.
+            match self.fenceable_token.generate_unfenced_token(self.mode)? {
+                Some((fence_updates, current_fenceable_token)) => {
+                    updates.extend(fence_updates.clone());
```

Comment:
I guess it doesn't matter because the `compare_and_append` is atomic (and maybe the updates even get sorted before applied?) but I do find it a little funny that you can smoosh the fence updates at the _end_ of what you're writing rather than at the beginning.

Reply:
Yeah, the order of the updates don't matter. They're all atomically applied at the same timestamp.